beastling/sections.py

```diff
@@ -1,33 +1,94 @@
+import sys
 from configparser import ConfigParser
+import pathlib
+import collections
+import functools
 
 import attr
 
+from beastling.util.misc import sanitise_tree
+
 __all__ = ['Admin']
 
 
+ConfigValue = collections.namedtuple('ConfigValue', ['value', 'fname'])
+
+
 @attr.s
 class Section(object):
     name = attr.ib()
-    cli_params = attr.ib(attr.Factory(dict))
+    cli_params = attr.ib(default=attr.Factory(dict))
+    files_to_embed = attr.ib(default=attr.Factory(set))
 
     @classmethod
     def from_config(cls, cli_params, section, cfg):
-        kw = {}
+        kw, files_to_embed = {}, set()
         for field in attr.fields(cls):
-            if field.name not in ['name', 'cli_params']:
+            if field.name not in ['name', 'cli_params', 'files_to_embed']:
                 opt = field.name
                 if opt.endswith('_'):
                     opt = opt[:-1]
                 if (section in cfg) and (opt in cfg[section]):
                     method = field.metadata.get('getter', ConfigParser.get)
-                    kw[field.name] = method(cfg, section, opt)
-        return cls(name=section, cli_params=cli_params, **kw)
+                    res = method(cfg, section, opt)
+                    if isinstance(res, ConfigValue):
+                        kw[field.name] = res.value
+                        files_to_embed.add(res.fname)
```

Comment:
With a bit of hacking we can even get the `files_to_embed` functionality in. The `Configuration` class would then have to aggregate the `files_to_embed` from all sections, but that doesn't change much.